cpp/ql/test/library-tests/deduction_guides/test.cpp

```diff
@@ -0,0 +1,18 @@
+// semmle-extractor-options: -std=c++20
```

Comment:
I'd like to see some more tests:
 1. If we don't use CTAD, are (implicit) deduction guides still emitted?
 2. If we have multiple implicit guides, are they all emitted, or only the ones that are used?
 3. If we have multiple explicit guides, are they all emitted, or only the ones that are used?

For 2+3, I think we can just extend the existing test to add an unused two-argument constructor and an unused two-argument guide. 1 would need to be a separate test.

Reply:
> 1. If we don't use CTAD, are (implicit) deduction guides still emitted?

I don't believe so, because that would have shown up in many other tests already, and it generally doesn't. Note that this is not related to what I'm doing here, as I'm only adding a relation between deduction guides and the class templates. Not adding more deduction guides to the database.

> 2. If we have multiple implicit guides, are they all emitted, or only the ones that are used?
> 3. If we have multiple explicit guides, are they all emitted, or only the ones that are used?

Would you be able to spell out some examples? That's probably the fastest way to get this done.

Reply:
Take the test added in this PR and add:
 2. A constructor `C(int, int)`
 3. A guide `C(char, char) -> C<char>`

Reply:
Added 2 & 3.